resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/DriverConfigurationStepsOrchestrator.scala

```diff
@@ -99,40 +99,77 @@ private[spark] class DriverConfigurationStepsOrchestrator(
         Option(new PythonStep(mainPyResource, additionalPythonFiles, filesDownloadPath))
       case _ => Option.empty[DriverConfigurationStep]
     }
-    val initContainerBootstrapStep = if ((sparkJars ++ sparkFiles).exists { uri =>
-      Option(Utils.resolveURI(uri).getScheme).getOrElse("file") != "local"
-    }) {
-      val initContainerConfigurationStepsOrchestrator =
-          new InitContainerConfigurationStepsOrchestrator(
-              namespace,
-              kubernetesResourceNamePrefix,
-              sparkJars,
+
+    val (localFilesDownloadPath, submittedDependenciesBootstrapSteps) =
+      if (areAnyFilesNonContainerLocal(sparkJars ++ sparkFiles)) {
+        val (submittedLocalFilesDownloadPath,
+            sparkFilesResolvedFromInitContainer,
+            mountSmallFilesWithoutInitContainerStep) =
+          // If the resource staging server is specified, submit all local files through that.
+          submissionSparkConf.get(RESOURCE_STAGING_SERVER_URI).map { _ =>
+            (filesDownloadPath, sparkFiles, Option.empty[DriverConfigurationStep])
+          }.getOrElse {
+            // Else - use a small files bootstrap that submits the local files via a secret.
+            // Then, indicate to the outer block that the init-container should not handle
+            // those local files simply by filtering them out.
+            val sparkFilesWithoutLocal = KubernetesFileUtils.getNonSubmitterLocalFiles(sparkFiles)
+            val smallFilesSecretName = s"$kubernetesAppId-submitted-files"
```

Comment:
might be worth wrapping that variable to `s"${kubernetesAppId}-submitted-files"` for clarity on what's expanded